# HG changeset patch
# User Michael Goffioul <michael.goffioul@gmail.com>
# Date 1242935317 -3600
#      Thu May 21 20:48:37 2009 +0100
# Node ID 7319e4de9e446195aaf5b7749cf65a8f4566560d
# Parent  8a5a04b6c46b06b35cb3f53a93c2fbdcc9da1967
Add missing return statement in base-aepbal.h

diff --git a/liboctave/ChangeLog b/liboctave/ChangeLog
--- a/liboctave/ChangeLog
+++ b/liboctave/ChangeLog
@@ -1,8 +1,13 @@
+2009-05-21  Michael Goffioul  <michael.goffioul@gmail.com>
+
+	* base-aepbal.h (base_aepbal::operator=): Add missing return
+	statement.
+
 2009-05-20  Jaroslav Hajek  <highegg@gmail.com>
 
 	* eigs-base.cc (
 	EigsRealSymmetricMatrix,
 	EigsRealSymmetricMatrixShift,
 	EigsRealSymmetricFunc,
 	EigsRealNonSymmetricMatrix,
 	EigsRealNonSymmetricMatrixShift,
diff --git a/liboctave/base-aepbal.h b/liboctave/base-aepbal.h
--- a/liboctave/base-aepbal.h
+++ b/liboctave/base-aepbal.h
@@ -45,16 +45,17 @@ public:
 
   base_aepbal& operator = (const base_aepbal& a)
     {
       balanced_mat = a.balanced_mat;
       scale = a.scale;
       ilo = a.ilo;
       ihi = a.ihi;
       job = a.job;
+      return *this;
     }
 
   MatrixT balanced_matrix (void) const { return balanced_mat; }
 
   VectorT permuting_vector (void) const
     {
       octave_idx_type n = balanced_mat.rows ();
       VectorT pv (n);
